src/stores/UserStore.ts

```diff
@@ -38,16 +38,23 @@ export class UserStore {
   }
 
   signOut = async () => {
-    await auth.signOut();
-    if (auth.currentUser == null) this._isNewUser = undefined;
+    await this._auth.signOut();
+    if (this._auth.currentUser == null) {
+      console.log(this._auth.currentUser);
+      this._isNewUser = undefined;
+    }
+  };
+
+  refreshUser = async () => {
+    await this._checkAuth();
   };
 
   //TODO: 함수명 맘에 안듬
   private _checkAuth = async () => {
     if (this._auth.currentUser != null) {
-      await this._fetchCurrentUser(this._auth.currentUser.uid);
+      await this._fetchCurrentUser(this._auth.currentUser!.uid);
```

Comment:
if 문에서 null check를 했기 때문에 smart cast가 되어 `!` 연산자는 없어도 될것같아요.
```suggestion
      await this._fetchCurrentUser(this._auth.currentUser.uid);
```